src/pudl/analysis/record_linkage/name_cleaner.py

```diff
@@ -5,37 +5,33 @@
 import logging
 import re
 from importlib.resources import files
-from typing import Literal
+from typing import Self
 
 import pandas as pd
-from pydantic import BaseModel
+from pydantic import BaseModel, Field, model_validator
 
 logger = logging.getLogger(__name__)
 
 CLEANING_RULES_DICT = {
     "remove_email": [" ", r"\S*@\S*\s?"],
     "remove_url": [" ", r"https*\S+"],
-    "remove_word_the_from_the_end": [" ", r"the$"],
-    "remove_word_the_from_the_beginning": [" ", r"^the"],
-    "place_word_the_at_the_beginning": [" ", r"the$"],
+    "remove_word_the_from_the_end": ["", r"\s+the$"],
+    "remove_word_the_from_the_beginning": ["", r"^the\s+"],
     "remove_www_address": [" ", r"https?://[.\w]{3,}|www.[.\w]{3,}"],
     "enforce_single_space_between_words": [" ", r"\s+"],
-    "replace_amperstand_by_AND": [" and ", r"&"],
-    "add_space_between_amperstand": [" & ", r"&"],
+    "replace_ampersand_by_AND": [" and ", r"\s*&\s*"],
+    "add_space_between_ampersand": [" & ", r"&"],
     "add_space_before_opening_parentheses": [" (", r"\("],
     "add_space_after_closing_parentheses": [") ", r"\)"],
-    "replace_amperstand_between_space_by_AND": [" and ", r"\s+&\s+"],
-    "replace_hyphen_by_space": [" ", r"-"],
-    "replace_hyphen_between_spaces_by_single_space": [" ", r"\s+-\s+"],
-    "replace_underscore_by_space": [" ", r"_"],
-    "replace_underscore_between_spaces_by_single_space": [" ", r"\s+_\s+"],
+    "replace_hyphen_by_space": [" ", r"\s*-\s*"],
+    "replace_underscore_by_space": [" ", r"\s*_\s*"],
     "remove_all_punctuation": [" ", r"([^\w\s])"],
     "remove_punctuation_except_dot": [" ", r"([^\w\s.])"],
     "remove_mentions": [" ", r"@\S+"],
     "remove_hashtags": [" ", r"#\S+"],
     "remove_numbers": [" ", r"\w*\d+\w*"],
-    "remove_text_puctuation": [" ", r'\;|\:|\,|\.|\?|\!|"'],
-    "remove_text_puctuation_except_dot": [" ", r'\;|\:|\,|\?|\!|"'],
+    "remove_text_punctuation": ["", r'\;|\:|\,|\.|\?|\!|"|\''],
+    "remove_text_punctuation_except_dot": [" ", r'\;|\:|\,|\?|\!|"\''],
```

Comment:
Should this replace with an empty string like the one above?

Reply:
yep, fixed